upgrade/scripts/upgrade/prerequisites.sh

```diff
@@ -544,26 +544,26 @@ if [[ ${state_recorded} == "0" && $(hostname) == "ncn-m001" ]]; then
     if [[ ${k8s_done} = 1 && ${ceph_done} = 1 ]]; then
         echo "Already ran ${NCN_IMAGE_MOD_SCRIPT}, skipping re-run."
     else
-        rm -f "${artdir}/storage-ceph/secure-storage-ceph-${CEPH_VERSION}.squashfs" "${artdir}/kubernetes/secure-kubernetes-${KUBERNETES_VERSION}.squashfs"
+        rm -f "${artdir}/storage-ceph/secure-storage-ceph-${CEPH_VERSION}-$(uname -i).squashfs" "${artdir}/kubernetes/secure-kubernetes-${KUBERNETES_VERSION}-$(uname -i).squashfs"
         DEBUG=1 "${NCN_IMAGE_MOD_SCRIPT}" \
             -d /root/.ssh \
-            -k "${artdir}/kubernetes/kubernetes-${KUBERNETES_VERSION}.squashfs" \
-            -s "${artdir}/storage-ceph/storage-ceph-${CEPH_VERSION}.squashfs" \
+            -k "${artdir}/kubernetes/kubernetes-${KUBERNETES_VERSION}-$(uname -i).squashfs" \
+            -s "${artdir}/storage-ceph/storage-ceph-${CEPH_VERSION}-$(uname -i).squashfs" \
             -p
     fi
 
     set -o pipefail
     IMS_UPLOAD_SCRIPT=$(rpm -ql docs-csm | grep ncn-ims-image-upload.sh)
 
-    export IMS_ROOTFS_FILENAME="${artdir}/kubernetes/secure-kubernetes-${KUBERNETES_VERSION}.squashfs"
-    export IMS_INITRD_FILENAME="${artdir}/kubernetes/initrd.img-${KUBERNETES_VERSION}.xz"
-    export IMS_KERNEL_FILENAME="${artdir}/kubernetes/*.kernel"
+    export IMS_ROOTFS_FILENAME="${artdir}/kubernetes/secure-kubernetes-${KUBERNETES_VERSION}-$(uname -i).squashfs"
+    export IMS_INITRD_FILENAME="${artdir}/kubernetes/initrd.img-${KUBERNETES_VERSION}-$(uname -i).xz"
+    export IMS_KERNEL_FILENAME="${artdir}/kubernetes/*-$(uname -i).kernel"
```

Comment:
this uname won't give correct output during an upgrade, will it? it'll be the current/old version.

Reply:
Correct me if I'm wrong, but one can not upgrade CSM from one architecture to another architecture.
I think the only way is if the target node was swapped out mid-way through its rebuild.

Reply:
Specifically CSM NCNs.
I don't think we even plan on supporting multi-arch clusters (e.g. arm and x86_64 ceph nodes or k8s nodes mixed together).

Reply:
yeah, totally. I missed that it was just doing the arch bit and not any version info.